dffml/df/base.py

```diff
@@ -215,6 +217,38 @@ def wrap(func):
                             param.annotation, param.annotation.__name__
                         ),
                     )
+                elif is_dataclass(param.annotation) or bool(
+                    issubclass(param.annotation, tuple)
+                    and hasattr(param.annotation, "_asdict")
+                ):
+                    # If the annotation is either a dataclass or namedtuple
+                    kwargs["inputs"][name] = Definition(
+                        name=".".join(name_list),
+                        primitive="map",
+                        spec=param.annotation,
+                    )
+                elif (
+                    get_origin(param.annotation) is list
+                    or get_origin(param.annotation) is dict
+                ):
+                    # If the annotation are of the form List[MyDataClass] or Dict[Any, MyDataClass]
+                    if get_origin(param.annotation) is list:
+                        primitive = "array"
+                        innerclass = list(get_args(param.annotation))[0]
+                    else:
+                        primitive = "map"
+                        innerclass = list(get_args(param.annotation))[1]
+
+                    if is_dataclass(innerclass) or bool(
+                        issubclass(innerclass, tuple)
+                        and hasattr(innerclass, "_asdict")
+                    ):
+                        kwargs["inputs"][name] = Definition(
+                            name=".".join(name_list),
+                            primitive=primitive,
+                            spec=innerclass,
+                            subspec=True,
+                        )
```

Comment:
Very nice! Can we add unit tests for `op` to check that this is working as expected?

Reply:
Or even just add example to the docstring of `op`, they'll get tested, that'd be even better

Reply:
The docstring of `Definition` might be helpful: https://intel.github.io/dffml/master/api/df/types.html#dffml.df.types.Definition

Reply:
Added the docstring examples, thanks